plugins/generic/usageStats/UsageStatsPlugin.inc.php

```diff
@@ -321,6 +380,31 @@ function _writeUsageEventInLogFile($usageEvent) {
 		}
 		fclose($fp);
 	}
+
+	//
+	// Private helper methods.
+	//
+	/**
+	* Hash (SHA256) the given IP using the given SALT.
+	*
+	* NB: This implementation was taken from OA-S directly. See
+	* http://sourceforge.net/p/openaccessstati/code-0/3/tree/trunk/logfile-parser/lib/logutils.php
+	* We just do not implement the PHP4 part as OJS dropped PHP4 support.
```

Comment:
We haven't officially dropped support for PHP4, but I think it's a non-issue by now. Especially in plugins, which can have their own requirements. OK to proceed.